src/kagglehub/clients.py

```diff
@@ -44,6 +44,44 @@
 """
 
 KAGGLEHUB_USER_AGENT = {"User-Agent": f"kagglehub/{kagglehub.__version__}"}
+_cached_user_agent = None
+
+
+class KaggleHubUserAgent:
+    def __init__(self) -> None:
+        self._cached_user_agent: Optional[str] = None
+
+    def _is_in_kaggle_notebook(self) -> bool:
+        return os.getenv("KAGGLE_NOTEBOOK_ENV_VAR_NAME") is not None
```

Comment:
This is wrong. You are checking for the Python variable name, not the name of the environment variable. Also, you must check the KAGGLE_DATA_PROXY_URL_ENV_VAR_NAME (this is the Python var name, not the name of the env variable).

Also, like we discussed in DMs, please refactor this to share the logic with the `ModelKaggleCacheResolver`. You could move this to a new `env.py` file which would include a `is_in_kaggle_notebook()` method.